dbms/src/Columns/Collator.h

```diff
@@ -3,9 +3,38 @@
 #include <string>
 #include <vector>
 #include <boost/noncopyable.hpp>
+#include <memory>
+#include <mutex>
 
 struct UCollator;
 
+/// Class represents available locales for collations.
+class AvailableCollationLocales : private boost::noncopyable
+{
+public:
+
+    struct LocaleAndLanguage
+    {
+        std::string locale_name; /// in ISO format
+        std::string language; /// in English
+    };
+
+    static AvailableCollationLocales & instance();
+
+    /// Get all collations with names
+    const std::vector<LocaleAndLanguage> & getAvailableCollations() const;
```

Comment:
Maybe using for `std::vector<LocaleAndLanguage>`?

Reply:
I think it's clear here.